page/jquery-ui/widget-factory/classes-option.md

```diff
@@ -0,0 +1,134 @@
+
+```
+dialog.dialog( "option", "classes.ui-dialog", "" );
+```
+
+And the following creates a [widget extension](http://learn.jquery.com/jquery-ui/widget-factory/extending-widgets/) that automatically associates the `custom-red` and `custom-big` class names with the `ui-dialog` class name:
+
+```
+<style>
+	.custom-red { background: red; }
+	.custom-big { font-size: 3em; }
+</style>
+<script>
+	$.widget( "custom.dialog", $.ui.dialog, {
+		options: {
+			classes: {
+				"ui-dialog": "custom-red custom-big"
+			}
+		}
+	});
+	$( "<div>Big and red</div>" ).dialog();
+</script>
+```
+
+As an added benefit, the widget factory also removes any class names specified in the `classes` option when the widget is destroyed.
+
+## Theming
+
+As the previous examples show, the `classes` option provides a quick way to associate custom class names with class names used within a widget. This approach works for simple cases, but it can also be used to adapt third-party themes to work with widget-factory-built widgets. For example, if you're using [Bootstrap](http://getbootstrap.com/) and jQuery UI together, you can use the following code to create a jQuery UI dialog that uses Bootstrap's theming:
+
+```
+$.extend( $.ui.dialog.prototype.options.classes, {
+	"ui-dialog": "modal-content",
+	"ui-dialog-titlebar": "modal-header",
+	"ui-dialog-title": "modal-title",
+	"ui-dialog-titlebar-close": "close",
+	"ui-dialog-content": "modal-body",
+	"ui-dialog-buttonpane": "modal-footer"
+});
+```
+
+For more examples of this approach, check out [Alexander Schmitz's repo](https://github.com/arschmitz/jqueryui-bootstrap-adapter) that adapts jQuery UI to work with Boostrap using the `classes` option.
+
+## Using the `classes` option in your own widgets
+
+Custom widgets can take advantage of the `classes` option just like the jQuery UI widgets do. In fact, the widget factory provides a number of new helper methods to make implementing your own `classes` option relatively easy. As an example consider the following custom redBox widget, *before* the `classes` option:
+
+```
+<style>
+	.custom-red-box { background: red; }
+</style>
+<script>
+	$.widget( "custom.redBox", {
+		_create: function() {
+			this.element.addClass( "custom-red-box ui-corner-all" );
+		},
+		_destroy: function() {
+			this.element.removeClass( "custom-red-box ui-corner-all" );
+		}
+	});
+</script>
+```
+
+Adding the `classes` option to this widget allows users to do two things: associate new class names with the `custom-red-box` class name, as well as control how the `ui-corner-all` class name is used. The code below shows an updated implementation of the `redBox` widget that makes these use cases possible:
+
+```
+<style>
+	.custom-red-box { background: red; }
+</style>
+<script>
+	$.widget( "custom.redBox", {
+		options: {
+			classes: {
+				"custom-red-box": "ui-corner-all"
+			}
+		},
+		_create: function() {
+			this._addClass( "custom-red-box" );
+		},
+		_destroy: function() {
```

Comment:
Remove the destroy method here completely its no longer needed since all you are doing is removing classes.